tools/sof_ri_info/sof_ri_info.py

```diff
@@ -0,0 +1,883 @@
+                                           offset)
+
+    def dump_info(self, pref):
+        print('{}{} type {} length {}'.
+              format(pref, self.name,
+                     self.adir['type'], self.adir['length']))
+
+class PlatFwAuthExtension(MftExtension):
+    """ Platform FW Auth Extension
+    """
+    def __init__(self, ext_id, offset):
+        super(PlatFwAuthExtension,
+              self).__init__(ext_id, 'Plat Fw Auth Extension', offset)
+
+    def dump_info(self, pref):
+        print('{}{}'.format(pref, self.name))
+
+class AdspMetadataFileExt(MftExtension):
+    """ ADSP Metadata File Extension
+    """
+    def __init__(self, ext_id, offset):
+        super(AdspMetadataFileExt,
+              self).__init__(ext_id, 'ADSP Metadata File Extension',
+                             offset)
+
+    def dump_info(self, pref):
+        out = '{}{}'.format(pref, self.name)
+        out += ' ver {}'.format(self.adir['version'])
+        out += ' base offset {}'.format(self.adir['base_offset'])
+        out += ' limit offset {}'.format(self.adir['limit_offset'])
+        print(out)
+        print('{}  IMR type {}'.format(pref, self.adir['adsp_imr_type']))
+        print('{}  Attributes'.format(pref))
+        print('{}    {}'.format(pref, self.adir['attributes']))
+
+class AdspManifest(Component):
+    """ ADSP Manifest
+    """
+    def __init__(self, name, offset):
+        super(AdspManifest, self).__init__('adsp_mft', name, offset)
+
+    def dump_info(self, pref):
+        hdr = self.cdir['adsp_mft_hdr']
+        out = '{}{} (ADSP Manifest)'.format(pref, self.name)
+        out += ' name {}'.format(hdr.adir['name'])
+        out += ' build ver {}'.format(hdr.adir['build_version'])
+        out += ' feature mask {}'.format(hdr.adir['feature_mask'])
+        out += ' image flags {}'.format(hdr.adir['fw_image_flags'])
+        print(out)
+        print('{}  HW buffers base address {} length {}'.
+              format(pref,
+                     hdr.adir['hw_buf_base_addr'],
+                     hdr.adir['hw_buf_length']))
+        print('{}  Load offset {}'.format(pref,
+                                          hdr.adir['load_offset']))
+        self.dump_comp_info(pref, 'ADSP Manifest Header')
+
+class FwBin(Component):
+    """ Parsed sof binary
+    """
+    def __init__(self):
+        super(FwBin, self).__init__('bin', 'SOF Binary', 0)
+
+    def dump_info(self, pref):
+        """ Print out the content
+        """
+        print('SOF Binary {} size {}'.format(
+            self.adir['file_name'], self.adir['file_size']))
+        self.dump_comp_info(pref)
+
+def main(args):
+    """ main function
+    """
+    Attribute.no_colors = args.no_colors
+    Attribute.full_bytes = args.full_bytes
+
+    fw_bin = parse_fw_bin(args.sof_ri_path, args.verbose, not args.headers)
+    fw_bin.dump_info('')
+
+if __name__ == "__main__":
+    ARGS = parse_params()
```

Comment:
Why is local variable uppercase?

Reply:
I was confused by this one as well, and they said Python was a clear and readable language 😅